fakestorage/server.go

```diff
@@ -268,6 +268,18 @@ func (s *Server) buildMuxer() {
 	s.mux.Host("{bucketName:.+}").Path("/{objectName:.+}").Methods(http.MethodPost, http.MethodPut).HandlerFunc(jsonToHTTPHandler(s.insertObject))
 }
 
+// publicHostMatcher matches incoming requests against the currently specified server publicHost.
+func (s *Server) publicHostMatcher(r *http.Request, rm *mux.RouteMatch) bool {
+	return r.Host == s.publicHost
+}
+
+// bucketHostMatcher matches incoming requests against the bucket subdomain of the
+// currently specified server publicHost.
+func (s *Server) bucketHostMatcher(r *http.Request, rm *mux.RouteMatch) bool {
+	bucketHost := fmt.Sprintf("{bucketName}.%s", s.publicHost)
+	return r.Host == bucketHost
```

Comment:
Should this be `return r.Host == bucketHost || r.host == s.publicHost`? Because we want `storage.googleapis.com` and `<bucket>.storage.googleapis.com` to work?

Reply:
(and then whatever the user sets it to)

Reply:
i actually just removed that method. this was a direct replacement for the just the bucket subdomain routes. each one of those routes already had an equivalent non-subdomain route.

however, one interesting this is that tests failed when i converted the subdomain routes. the tests that test those routes actually suffer from the same problem as this fixes for use with containers: the test servers are defined like `NewServerWithOptions(Options{PublicHost: "127.0.0.1"})`, missing the port, because the port is only known after the httptest server has started.